### ISSUE TITLE
429 Error

### ISSUE THREAD

**Issue by Francopacks** [brbeaird/SmartThings-MyQ-Edge#48]
![Screenshot_20231018_124137_Remote Ripple](https://github.com/brbeaird/SmartThings-MyQ-Edge/assets/88859118/c4793eaf-8dc8-4b05-bd57-2522ec9a2cc8)
 Consistent 429 errors even after leaving off overnight.

**Comment by Francopacks** [comment]
I am on 1.0.6

**Comment by danman91** [comment]
I have also had same issue for a couple days.  Was happening on 1.0.4 but continued after update to 1.0.6.

**Comment by brbeaird** [comment]
A few other people are reporting issues again around other integrations. May need another update soon.

**Comment by samuelskwong** [comment]
Having the same problem on my docker container - verified that I could login fine on my MyQ phone app

**Comment by johnlinehan** [comment]
Same problem - also on the latest version.

**Comment by brbeaird** [comment]
It's a mixed bag right now, both for SmartThings and HomeAssistant users. For some people, it's still working fine. For others, it has started working again after a long delay. Still for others, they are still locked out. I'm keeping an eye out for what people figure out.

**Comment by samuelskwong** [comment]
Just tried again now w/o upgrading or changing anything and was able to connect successfully with ver 1.0.6.  It really may just be a timing thing.

I was not able to authenticate since 10/17 - continued on for almost 3 days.

**Comment by Francopacks** [comment]
I was also able to connect to my devices just now.  Wierd.

**Comment by Francopacks** [comment]
![Screenshot_20231021_122841_Remote Ripple](https://github.com/brbeaird/SmartThings-MyQ-Edge/assets/88859118/eecc9605-5793-4ee2-a3a9-3c859ceee7be)

**Comment by gnieboer** [comment]
I was on 1.0.3 and was working fine until a couple days ago then 429's including this morning.  Just upgraded to 1.0.6 and it's fine now.

**Comment by TTVert** [comment]
I was on 1.0.4 and upgraded to 1.0.6 to see if I could login but also got 429.  I hoped that perhaps I could wipe and re-enter my creds in smart things under the MyQ-Controller but no luck there :(. 

This is what smart things is showing if you care.
![image](https://github.com/brbeaird/SmartThings-MyQ-Edge/assets/24374538/1127b56e-0306-435b-a0e9-532b228cc591)
Dave

**Comment by johnlinehan** [comment]
Mine just magically reconnected after days of the same error. Back in business

**Comment by JanVanHyfte** [comment]
I was yesterday just setting up this nice piece of artwork, and after 1 successful communication (could open and close port), the damn thing started reporting 429's....

**Comment by JanVanHyfte** [comment]
I was yesterday just setting up this nice piece of artwork, and after 1 succesfull communication (could open and close port), the damn thing started reporting 429's....

**Comment by TTVert** [comment]
Started working for me at 7:33PM UTC.

Dave

**Comment by neilschelly** [comment]
I'm curious if the folks who see it just start working out of nowehere are leaving it running until it works or if you are leaving it off for awhile and starting it up the next day?

I've been leaving it running, hoping it just changes to working. The 429 sounds a bit like maybe too many failures from an IP causes it to get blocked, so maybe it needs a long break. Or maybe it's entirely a client-side bug and I've overthinking the error message...
-N

**Comment by Francopacks** [comment]
Leaving it off for a few days seemed to work for me.

It worked for about 24 hours and is now back to doing 429s

**Comment by JanVanHyfte** [comment]
I left it active in my docker container, and this afternoon it suddenly "woke up/stopped throwing 429's", and now it is working ...

**Comment by velojunkie** [comment]
I purged Docker from my rapsberry pi and setup everything from scratch two days ago. It worked initially then threw a 429 a few hours later so I just let it sit active in Docker. I noticed last night, about 24 hours later, that it's 'magically' connected again.

**Comment by Xtropy74** [comment]
Likely a problem on their end based on the intermittent nature of the issue

**Comment by jsalh** [comment]
I have the Windows version and shut it down for 24 hrs .. and when I restarted the service this morning everything started to work.  
However, checked it now (8 hours later), and the once again the 429 issue has comeback.

**Comment by TTVert** [comment]
> I purged Docker from my rapsberry pi and setup everything from scratch two days ago. It worked initially then threw a 429 a few hours later so I just let it sit active in Docker. I noticed last night, about 24 hours later, that it's 'magically' connected again.

I may move this container to my Pi, how complicated was it to do this?   Did you do it all from CLI or GUI?  I don't even think I have a GUI on this build.
Thanks
Dave

**Comment by TTVert** [comment]
> > I may move this container to my Pi, how complicated was it to do this? Did you do it all from CLI or GUI? I don't even think I have a GUI on this build. Thanks Dave
> 
> It's super simple and I use PuTTY for SSH connect (no gui for me either). Just install Docker, pull down the image, and then start the container. brbeaird's instructions are really good.
> 
> As a side note: My-QController is still connected so looks like the issue has been resolved.
Thanks, i'm going to give it a go.  I'd rather get this off my windows machine if possible.

Dave

**Comment by trevorkwhite** [comment]
Same deal here, seeing wider reports on Reddit as well on this - seems widespread at the moment and intermittent. Upgraded to the latest container as well, still 429 (also increased polling to 30s)

**Comment by bds001** [comment]
I'm having the same issue now. Started about 2 days ago.  I've tried turning off the bridge for a day and then logging in again but still no success.

**Comment by trevorkwhite** [comment]
> So What’s going on now?

MyQ has enabled [cloudflare bot protection 32](https://www.cloudflare.com/application-services/products/bot-management/) which uses Machine Learning to determine what is a bot and what isn’t.

This makes the task much harder to overcome. There are some changes that seem promising, but the issue is once I make said changes and everyone updates their myq instance, the bot protection will pick that up and myq will again be broken - this is partially why some other implementations of the myq api are still working, they have a smaller user count and the bot protection has not picked up on it (yet).

Let me make a few things clear, this has nothing to do with rate limiting. I have seen a lot of users talk about how we should implement a option to change how frequently we poll for data, or just don’t use the api as much as we do now. The issue is only with authenticating - that is the only part that the anti bot protection is enabled. It does not matter how frequently you get data from myq, as soon as you need to reauth yourself, you can end up getting a 429 error - it is not a 429 error in the traditional sense.

I am in discussion with a few other third party myq developers and we will attempt to fix the problem, but it does not seem like there is an easy solution. I am primarily talking with the developer of the homebridge version of myq who has been maintaining that api for 10 years.

MyQ is explicitly attempting to block third party integrations, and when it comes down to it - when you have a team of software engineers getting paid to block us vs people attempting to reverse engineer the api in their free time, the paid software engineers are going to win over the long term.

https://community.home-assistant.io/t/the-current-state-of-myq-from-the-codeowner/630623

This thread has a ton of useful info ^^

**Comment by brbeaird** [comment]
That post is good, but I remain more optimistic at the moment. The bot detection is only aggressively targeting the initial login page, so if I can find a good way for people to get an initial token, I can modify the process to do a better job of reusing that token at all costs. Currently, the library is very sensitive to any error and tends to bail out to the main login page more often than it should, especially when that's the fast-track to getting cloudflare challenged. I'll keep updates posted to the SmartThings community thread.

**Comment by trevorkwhite** [comment]
Posting that thread here since I had to look it up again (latest post as of this morning) - https://community.smartthings.com/t/st-edge-smartthings-myq-edge/248988/1508?u=trevorwh

**Comment by brbeaird** [comment]
I am sorry to say this integration is officially dead, at least for the time being. The API changes by MyQ are dramatic and are not possible to overcome at this point. I suggest looking at alternatives like Meross. I also have a Ratgdo device on order - once I get my hands on it, I may see if I can get that integrated with SmartThings as a replacement.

**Comment by TTVert** [comment]
Well that's very sad.

**Comment by brbeaird** [comment]
More details here - https://www.theverge.com/23949612/chamberlain-myq-smart-garage-door-controller-homebridge-integrations

**Comment by velojunkie** [comment]
Thanks for all you've done with this, brbeaird. Really appreciate it! 

Keep us posted on Ratgdo; I've been looking at this too; planning for the day when my MyQ-Controller craps out.

EDIT: MyQ-Controller lost connectivity today, Nov 9. RIP.

**Comment by Francopacks** [comment]
Yeah, Thanks for all you've done over the years brbeaird!  We all really appreciate your efforts.

**Comment by slynnmurphy** [comment]
Mine stopped working today.  Thanks brbeaird for providing this and the pre-edge version.  I've enjoyed letting Alexa open and close the door for me.   And the pre-edge version provided an improvement over the base myQ door, because myQ would get confused and think the door was open when it was closed.  Being able to integrate a different position sensor (back when SmartThings Multipurpose could do that), allowed me to know the true state of the door.  Otherwise, the myQ app would let me "close" a door that wasn't really open, which actually opened it!  One of the main reasons I got myQ in the first place was to be able to close the door, if I forgot to close it.   But, I couldn't trust it to tell me the correct state of the door, so that didn't really work out.  It seems like it is somewhat better than when I first used it, but still not completely accurate.  I'll be looking for another solution...

**Comment by brbeaird** [comment]
Appreciate the kind words. I'm getting more excited about the possibilities of a new Edge driver with ratgdo. Good chance it could be a nice reboot of this project.

**Comment by TTVert** [comment]
Hey I know it’s been some time and that MY Q decided to screw us.  I noticed recently that w/ the IFTTT integration you can now chose to close the garage, I was wondering if that gives us any ability to perhaps reverse engineer that to actually open the garage as well?

Dave
